### ISSUE TITLE
Docs: The options of the ban-ts-comment rule seem to habe changed in v6 but I did not find it on the change logs

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#7239]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The options of rule `ban-ts-comment seems to have changed` seem to have changed in v6 but this breaking change does not seem to be documented in the announcement

### Affected URL(s)

https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/

**Comment by Josh-Cena** [comment]
There are some refactors happening, but there should not be behavior changes. What do you think has changed?

**Comment by doberkofler** [comment]
My original rule looked like this:
```
'@typescript-eslint/ban-ts-comment': ['error',
	{
		'ts-expect-error': {
			'allow-with-description': true,
			'descriptionFormat': '^ ts\\(\\d\\d\\d\\d\\) FIXME: .+$', // Use a format like this: "// @ts-expect-error ts(2367) FIXME: ..."
			'minimumDescriptionLength': 10,
		},
		'ts-ignore': true,
		'ts-nocheck': true,
		'ts-check': false,
	}
],
```
and it seems as if `allow-with-description`  and `minimumDescriptionLength` are no longer allowed in v6.

**Comment by Josh-Cena** [comment]
`minimumDescriptionLength` has never been a directive config. It should be root level. Similarly, `"allow-with-description"` is used by itself, not as a property. You should use this:

```js
'@typescript-eslint/ban-ts-comment': ['error',
	{
		'ts-expect-error': {
			'descriptionFormat': '^ ts\\(\\d\\d\\d\\d\\) FIXME: .+$', // Use a format like this: "// @ts-expect-error ts(2367) FIXME: ..."
		},
		'ts-ignore': true,
		'ts-nocheck': true,
		'ts-check': false,
		minimumDescriptionLength: 10,
	}
],
```

The fact that it used to work was a bug in our validation schema.

**Comment by bradzacher** [comment]
For reference with v6 we built tolling to help us visualise our options as TS types - which helped us find and fix some loose spots in some schemas - so config errors are expected!

**Comment by doberkofler** [comment]
Maybe a short note on fixes like this in the change log might help others but I completely understand that this is not a breaking change. Thank you for all the good work!
